web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,294 @@
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import {
+  Box,
+  ButtonPrimary,
+  ButtonText,
+  Card,
+  Flex,
+  Indicator,
+  Input,
+  LabelInput,
+  Text,
+} from 'design';
+import * as Alerts from 'design/Alert';
+import ButtonSso, { guessProviderType } from 'shared/components/ButtonSso';
+import { useRefAutoFocus } from 'shared/hooks';
+import { Attempt, useAsync } from 'shared/hooks/useAsync';
+import { AuthProvider } from 'shared/services';
+
+import ResourceService from 'teleport/services/resources';
+import { storageService } from 'teleport/services/storageService';
+
+type Props = {
+  onLoginWithSso(provider: AuthProvider): void;
+  /**
+   * onUseLocalLogin is called to switch the view to the local login form.
+   */
+  onUseLocalLogin(): void;
+};
+
+/**
+ * FormIdentifierFirst is the login form for identifier-first login.
+ */
+export function FormIdentifierFirst({
+  onLoginWithSso,
+  onUseLocalLogin,
+}: Props) {
+  const [resourceService] = useState(() => new ResourceService());
+
+  const [rememberedUsername, setRememberedUsername] = useState<string>(
+    storageService.getRememberedSsoUsername().trim()
+  );
+  const [username, setUsername] = useState<string>(rememberedUsername);
+  const [connectors, setConnectors] = useState<AuthProvider[]>([]);
+
+  useEffect(() => {
+    if (rememberedUsername) {
+      fetchMatchingConnectors(rememberedUsername);
+    }
+  }, [rememberedUsername]);
+
+  const [fetchAttempt, fetchMatchingConnectors] = useAsync(
+    useCallback(
+      async (username: string) => {
+        const matchedConnectors =
+          await resourceService.getUserMatchedAuthConnectors(username);
+        if (matchedConnectors.length === 0) {
+          if (rememberedUsername) {
+            // If we have a remembered username but no connectors, we clear the remembered username.
+            storageService.clearRememberedSsoUsername();
+            setRememberedUsername('');
+            setUsername('');
+            return;
+          }
+          throw new Error(`No SSO connectors found for user: ${username}`);
+        }
+        // If there isn't a remembered username, and there is only one matching connector, we take them straight to the IdP.
+        if (matchedConnectors.length === 1 && !rememberedUsername) {
+          onLoginWithSso(matchedConnectors[0]);
+          storageService.setRememberedSsoUsername(username);
+          setRememberedUsername(username);
+          return;
+        }
+        setConnectors(matchedConnectors);
+        setRememberedUsername(username);
+        storageService.setRememberedSsoUsername(username);
+        return;
+      },
+      [username]
+    )
+  );
+
+  const onSubmitUsername = () => {
+    fetchMatchingConnectors(username);
+  };
+
+  const onNotYou = () => {
+    storageService.clearRememberedSsoUsername();
+    setUsername('');
+    setRememberedUsername('');
+    setConnectors([]);
```

Comment:
Will we ever have a mechanism to remember multiple users? 
It makes is super easy for users if we support that. 
Example of azure portal
<img width="252" alt="Screenshot 2025-07-02 at 4 59 05 PM" src="https://github.com/user-attachments/assets/739930e7-433d-41d2-a72b-781cb3a56a82" />

Reply:
We have designs for it, but after discussing with Kenny we decided it wouldn't be very useful here and may actually make things more cumbersome since the vast majority of Teleport users will only ever use account.